File: java/src/main/java/br/com/rh/models/Outsourced.java
package br.com.rh.models;

import br.com.rh.enums.Role;

import java.math.BigDecimal;

public class Outsourced {
    private PersonalData personalData;
    private String enterprise;

    public Outsourced(String name, String cpf, Role role, BigDecimal salary, String enterprise) {
        this.enterprise = enterprise;
        this.personalData = new PersonalData(name, cpf, role, salary);
    }

    public String getEnterprise() {
        return enterprise;
    }

    public void setEnterprise(String enterprise) {
        this.enterprise = enterprise;
    }
}

File: typescript/src/models/outsourced.ts
import { Role } from "../enums/role.enum";
import { PersonalData } from "./personal-data";

export class Outsourced {
  private personalData: PersonalData;
  private enterprise: string;

  constructor(
    name: string,
    cpf: string,
    role: Role,
    salary: number,
    enterprise: string
  ) {
    this.enterprise = enterprise;
    this.personalData = new PersonalData(name, cpf, role, salary);
  }

  public getEnterprise() {
    return this.enterprise;
  }

  public getPersonalData() {
    return this.personalData;
  }
}
